src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+      topicsCall.mockClear();
+
+      await mirror.update(reporter);
+      const laterPosts = repo.posts();
+
+      expect(fetcher._topic(4)).toMatchObject({id: 4, categoryId: 4});
+      expect(fetcher._topic(5)).toMatchObject({id: 5, categoryId: 4});
+      expect(fetcher._post(p5)).toMatchObject({id: 5, cooked});
+      // The initial load has strict order requirements.
+      expect(earlyCalls).toEqual([[1], [2], [3], [5], [4]]);
+      // The update does not.
+      expect(topicsCall).toHaveBeenCalledTimes(2);
+      expect(topicsCall).toHaveBeenCalledWith(4);
+      expect(topicsCall).toHaveBeenCalledWith(5);
+      expect(laterPosts).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 1,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2001,
+            "topicId": 1,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 2,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2002,
+            "topicId": 2,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 3,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2003,
+            "topicId": 3,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 4,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2004,
+            "topicId": 4,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<p>New content, without bump</p>",
+            "id": 5,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2005,
+            "topicId": 5,
+          },
+        ]
+      `);
+    });
+
+    it("mirrors supports the uncategorized category in recheckTopicsInCategories option", async () => {
+      const {mirror, fetcher, reporter} = example({
+        recheckTopicsInCategories: [1],
+      });
+
+      // Adds a category description #4.
+      // And a topic 5 which is a member of the #4 category.
+      fetcher.addPost(1, null);
+      fetcher.addPost(2, null);
+
+      // See we're trying to update #4 and #5.
+      const topicsCall = jest.spyOn(fetcher, "topicWithPosts");
+      await mirror.update(reporter);
+      const earlyCalls = [...topicsCall.mock.calls];
+      topicsCall.mockClear();
+      await mirror.update(reporter);
+
+      // The initial load has strict order requirements.
+      expect(earlyCalls).toEqual([[1], [2]]);
+      // The update does not.
+      expect(topicsCall).toHaveBeenCalledTimes(2);
```

Comment:
Not readily clear to me what this test is checking.

Reply:
The fetcher should be called exactly twice in the second update.
Because there should be exactly two topics discovered in the first update that are in the category we're forcing updates of the second time.

Without this assert it could be more than 2, as the below checks only assert at least those two distinct calls have been made. If the actual calls were: `[1, 1, 2, 1, 2, 2, 44]` it would pass.
```
      expect(topicsCall).toHaveBeenCalledWith(1);
      expect(topicsCall).toHaveBeenCalledWith(2);
```